sql/migrations/schema-postgresql-move-to-mirrors.sql

```diff
@@ -96,3 +96,204 @@ AS $BODY$
+  SELECT identifier FROM server WHERE id=arg_serverid
+$BODY$;
+
+-- FUNCTION: public.mirr_get_name(smallint[])
+
+DROP FUNCTION IF EXISTS public.mirr_get_name(smallint[]);
+
+CREATE OR REPLACE FUNCTION public.mb_mirror_identifier(
+    ids smallint[]
+  )
+    RETURNS text[]
+    LANGUAGE 'sql'
+AS $BODY$
+  SELECT array_agg(identifier::text) FROM server WHERE id = ANY(ids);
+$BODY$;
+
+-- FUNCTION: public.mirr_get_nfiles(integer)
+
+DROP FUNCTION IF EXISTS public.mirr_get_nfiles(integer);
+
+CREATE OR REPLACE FUNCTION public.mb_mirror_filecount(
+	  arg_serverid integer
+  )
+    RETURNS bigint
+    LANGUAGE 'sql'
+AS $BODY$
+  SELECT count(*) FROM mirrors WHERE server_id = arg_serverid;
+$BODY$;
+
+-- FUNCTION: public.mirr_get_nfiles(text)
+
+DROP FUNCTION IF EXISTS public.mirr_get_nfiles(text);
+
+CREATE OR REPLACE FUNCTION public.mb_mirror_filecount(
+	  arg_server_identifier text
+  )
+    RETURNS bigint
+    LANGUAGE 'sql'
+AS $BODY$
+  SELECT count(*) FROM mirrors WHERE (SELECT id FROM server WHERE identifier = arg_server_identifier) = server_id;
+$BODY$;
+
+-- FUNCTION: public.mirr_hasfile_byid(integer, integer)
+
+DROP FUNCTION IF EXISTS public.mirr_hasfile_byid(integer, integer);
+
+CREATE OR REPLACE FUNCTION public.mb_mirror_hasfile(
+    arg_serverid integer,
+    arg_fileid integer
+  )
+    RETURNS boolean
+    LANGUAGE 'plpgsql'
+AS $BODY$
+DECLARE
+    result integer;
+BEGIN
+    SELECT INTO result 1 FROM mirrors WHERE filemetadata_id = arg_fileid AND server_id = arg_serverid;
+    IF result > 0 THEN
+        RETURN true;
+    END IF;
+    RETURN false;
+END;
+$BODY$;
+
+-- FUNCTION: public.mirr_hasfile_byname(integer, text)
+
+DROP FUNCTION IF EXISTS public.mirr_hasfile_byname(integer, text);
+
+CREATE OR REPLACE FUNCTION public.mb_mirror_hasfile(
+    arg_serverid integer,
+    arg_path text
+  )
+    RETURNS boolean
+    LANGUAGE 'plpgsql'
+AS $BODY$
+DECLARE
+    result integer;
+    fileid integer;
+BEGIN
+    SELECT INTO fileid id FROM filemetadata WHERE path = arg_path;
+
+    -- There are three cases to handle, and we want to handle each of them
+    -- with the minimal effort.
+    -- In any case, we return a file id in the end.
+    IF fileid IS NULL THEN
+        RAISE DEBUG 'we do not know about the file "%".', arg_path;
+    ELSE
+      SELECT INTO result 1 FROM mirrors WHERE filemetadata_id = fileid AND server_id = arg_serverid;
+      IF result > 0 THEN
+          RETURN true;
+      END IF;
+    END IF;
+    RETURN false;
+END;
+$BODY$;
+
+DROP VIEW IF EXISTS hexhash;
+CREATE VIEW hexhash AS
+  SELECT
+    id AS file_id,
+    mtime,
+    size,
+    md5,
+    encode(md5, 'hex') AS md5hex,
+    sha1,
+    encode(sha1, 'hex') AS sha1hex,
+    sha256,
+    encode(sha256, 'hex') AS sha256hex,
+    sha1piecesize,
+    sha1pieces,
+    encode(sha1pieces, 'hex') AS sha1pieceshex,
+    btih,
+    encode(btih, 'hex') AS btihhex,
+    pgp,
+    zblocksize,
+    zhashlens,
+    zsums,
+    encode(zsums, 'hex') AS zsumshex
```

Comment:
we need `path` column here as well, so the view can be queried without join

Reply:
I thought about it. but right know it is 2 queries anyway.
https://github.com/openSUSE/mirrorbrain/blob/opensuse/mod_mirrorbrain/mod_mirrorbrain.c#L139..L163

Reply:
I mean instead of subquery in the hash query https://github.com/openSUSE/mirrorbrain/blob/f33ca265bb221f6799eb43702bf340f51672a7ff/mod_mirrorbrain/mod_mirrorbrain.c#L157-L159 we can just have `WHERE path = %s` if the view would have `path`

Reply:
This is how til will look if the column is inside view https://github.com/openSUSE/mirrorbrain/pull/52/files#diff-a44eaf51cd9831129dc4b515db0bd2aeL157-R156